File: src/analyzer/format-var-diagnostic.test.ts
import { describe, expect, test } from "vitest";

import { DIAGNOSTIC_KIND } from "./diagnostic-kind.js";
import { formatVarDiagnostic } from "./format-var-diagnostic.js";

describe("formatVarDiagnostic", () => {
  test("renders a single-line warning with line:column and the diagnostic message", () => {
    const lines = formatVarDiagnostic({
      kind: DIAGNOSTIC_KIND.VarDetected,
      message: "var declaration detected; rendered as node only (no edges).",
      span: { line: 3, column: 0, offset: 11 },
    });
    expect(lines).toEqual([
      "uns: warning: L3:0: var declaration detected; rendered as node only (no edges).",
    ]);
  });
});

File: crates/unsnarl-analyzer/src/format_var_diagnostic_test.rs
use unsnarl_ir::diagnostic::Diagnostic;
use unsnarl_ir::diagnostic_kind::DiagnosticKind;
use unsnarl_ir::primitive::{SourceColumn, SourceLine, SourceOffset, Span};

use super::format_var_diagnostic;

#[test]
fn renders_single_line_warning_with_line_column_and_message() {
    let diagnostic = Diagnostic {
        kind: DiagnosticKind::VarDetected,
        message: "var declaration detected; rendered as node only (no edges).".to_string(),
        span: Span {
            line: SourceLine(3),
            column: SourceColumn(0),
            offset: SourceOffset(11),
        },
    };
    let lines = format_var_diagnostic(&diagnostic);
    assert_eq!(
        lines,
        vec![
            "uns: warning: L3:0: var declaration detected; rendered as node only (no edges)."
                .to_string()
        ]
    );
}
